src/Protocol.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef SONAR_OCULUS_M750D_PROTOCOL_HPP
+#define SONAR_OCULUS_M750D_PROTOCOL_HPP
+
+#include "Oculus.h"
+#include "SonarData.hpp"
+#include <base/samples/Sonar.hpp>
+#include <stdio.h>
+
+namespace sonar_oculus_m750d {
+    class Protocol {
+    public:
+        void handleBuffer(uint8_t const* buffer);
+        base::samples::Sonar parseSonar();
+        /**
+         * @brief Rearrange the sonar data in beam major order
+         * 
+         * Takes an sonar data in bin major order, this is
+         * [idx = bin * beam_count + beam], and returns an vector of data arranged
+         * in beam major order, this is [idx = beam * bin_count + bin]
+         * 
+         */
+        static std::vector<float> toBeamMajor(uint8_t* bin_first,
+            uint16_t beam_count,
+            uint16_t bin_count);
+        static base::Time binDuration(double range, double speed_of_sound, int bin_count);
+
+    private:
+        void handleMessagePingResult(uint8_t const* buffer);
+        void handleMessageSimplePingResult(uint8_t const* buffer,
+            OculusMessageHeader const& header);
+
+        SonarData m_data;
+        bool m_simple_ping_result = false;
+
+        static constexpr double HORIZONTAL_FOV_DEG = 130;
+        static constexpr double VERTICAL_FOV_DEG = 20;
```

Comment:
I would have preferred that this came from a property. We could later rename the repo to `sonar_oculus` and use it for other types of sonars from them, which would have different fovs. For now, keeping it like this is fine, you dont know if we are ever getting a new sonar from them, and if we do it might not even have an equivalent implemetation